pkg/buildid/buildid.go

```diff
@@ -28,43 +28,39 @@ import (
 	"github.com/parca-dev/parca-agent/pkg/elfreader"
 )
 
-func BuildID(path string) (string, error) {
-	f, err := elf.Open(path)
-	if err != nil {
-		return "", fmt.Errorf("failed to open elf: %w", err)
-	}
+type ElfFile struct {
+	Path string
```

Comment:
This is more explicit, for sure. However, you can also use https://pkg.go.dev/os#File.Name

`Name returns the name of the file as presented to Open.`

You can ensure the file is open with an absolute path and everything can be consistent. This is not even needed, we don't do any async work on the paths, and we don't change the running directory during the agent's lifetime.

Reply:
So I think we can eliminate this additional abstraction.

Reply:
Oh, I see. They don't embed the `os.File` even though they use it underneath. `debug/elf` is not the best and most eloquent Go package out there, for sure. How about using the existing `ObjectFile` struct and passing it around? I want to simplify things. Having all these ephemeral types confuses the developer.

How about converging everything on the `ObjectFile` type? Open it once and pass the instances of that around. If it's too much work, I'm fine with this. Nothing to do 🤷

Reply:
As you said, it is a long way. And the Go team is not very keen on accepting changes to these packages. But we can try.

Reply:
> Oh, I see. They don't embed the `os.File` even though they use it underneath. `debug/elf` is not the best and most eloquent Go package out there, for sure. How about using the existing `ObjectFile` struct and passing it around? I want to simplify things. Having all these ephemeral types confuses the developer.
> 
> How about converging everything on the `ObjectFile` type? Open it once and pass the instances of that around. If it's too much work, I'm fine with this. Nothing to do shrug

It is not a lot of work but will cause a cyclic dependency between the `objectfile` and `buildid` packages; having both in one package will complicate the api more rn imo.
How about we either
 - just pass `path` as a parameter right now(and add a TODO there) or
 - stay with this `ElfFile`

till we come up with a better api or wait and see if we can add a `Name` method in the stdlib?

Reply:
Sounds good. Let's go with the `ElfFile` approach.